packages/docs/docs/components/spline.mdx

```diff
@@ -0,0 +1,493 @@
+
+While animating splines isn't too different from animating any other node, this
+section aims to illustrate a few of the most common use cases.
+
+### Drawing splines
+
+Similar to the [`Line`][line] component, the `Spline` node provides `start` and
+`end` signals which allow us to control the segment of the curve that should be
+visible. Both `start` and `end` are values between `0` and `1` and represent the
+percentage of the spline's arclength from which to start drawing from.
+
+```tsx {7-8}
+<Spline
+  points={[
+    [-300, 0],
+    [-150, -100],
+    [150, 100],
+  ]}
+  start={0.4}
+  end={0.8}
+/>
+```
+
+The example above would draw the spline starting at 40% of the spline's
+arclength (`start={0.4}`) and draw it up until 80% of the spline's arclength
+(`end={0.8}`).
+
+:::note
+
+When using `start` and `end` in conjunction with `startOffset` and `endOffset`,
+`start` and `end` will be relative to the _remaining_ length of the spline after
+taking the offset into account.
+
+:::
+
+We can then animate drawing a spline by tweening these properties:
+
+```tsx editor
+export default makeScene2D(function* (view) {
+  const spline = createRef<Spline>();
+
+  view.add(
+    <Spline
+      ref={spline}
+      lineWidth={6}
+      stroke={'lightseagreen'}
+      points={[
+        [-300, 0],
+        [-150, -100],
+        [150, 100],
+        [300, 0],
+      ]}
+      end={0}
+    />,
+  );
+
+  yield* spline().end(1, 1.5);
+  yield* spline().start(1, 1.5).to(0.5, 1);
+  yield* spline().end(0.5, 1);
+  yield* all(spline().start(0, 1.5), spline().end(1, 1.5));
+});
+```
+
+### Animating the knots of a spline
+
+Since the `Knot` node is a regular Motion Canvas component, it can be animated
+in much the same way as other components.
+
+:::note
+
+Animating a spline's knots is only possible when using the
+[Knot](#using-knot-nodes) component, not when using the `points` property.
+
+:::
+
+Below are a few examples of interesting effects that can be achieved by
+animating different properties of knots.
+
+<Tabs>
+  <TabItem value='position' label='Position' default>
+
+```tsx editor
+export default makeScene2D(function* (view) {
+  const knotPositions: PossibleVector2[] = [
+    [-200, 0],
+    [-100, -80],
+    [0, 80],
+    [100, -80],
+    [200, 0],
+  ];
+  const knots: Knot[] = [];
+
+  view.add(
+    <Spline lineWidth={6} stroke={'lightseagreen'}>
+      {knotPositions.map((pos, i) => (
+        <Knot ref={makeRef(knots, i)} position={pos} />
+      ))}
+    </Spline>,
+  );
+
+  yield* all(
+    knots[1].position.y(80, 1).to(-80, 1),
+    knots[2].position.y(-80, 1).to(80, 1),
+    knots[3].position.y(80, 1).to(-80, 1),
+  );
+});
+```
+
+  </TabItem>
+
+  <TabItem value="rotation" label="Rotation">
+
+Since `startHandle` and `endHandle` are relative to the knot, rotating the knot
+will also change the handle positions.
+
+:::caution
+
+Note that animating a knot's `rotation` is only possible if its `auto` value is
+less than `1`.
+
+:::
```

Comment:
This information should be hoisted outside of the tabs since it applies to transform in general.
You could say:
```
You can think of `startHandle` and `endHandle` as being the children of the Knot -
changing the knot's position, rotation and scale will also transform the handles.
The only exceptions are `auto` handles which are unaffected by these transformations.
```